control/iosys.py

```diff
@@ -928,18 +930,21 @@ def __init__(self, syslist, connections=[], inplist=[], outlist=[],
             sysobj_list.append(sys)
             sysname_list.append(sys.name)
 
+        # This might cause a problem if there are duplicate names...
+        states = states or nstatenames
+
         # Create the I/O system
         super(InterconnectedSystem, self).__init__(
             inputs=len(inplist), outputs=len(outlist),
-            states=nstates, params=params, dt=dt)
+            states=states, params=params, dt=dt, name=name)
 
         # If input or output list was specified, update it
         nsignals, self.input_index = \
             self._process_signal_list(inputs, prefix='u')
         if nsignals is not None and len(inplist) != nsignals:
             raise ValueError("Wrong number/type of inputs given.")
         nsignals, self.output_index = \
-            self._process_signal_list(outputs, prefix='y')
+            self._process_signal_list(outputs or len(outlist), prefix='y')
```

Comment:
What is the purpose of this change and why was it not applied to `inputs` directly above? The function returns `None, {}` for `_process_signal_list(None)`

Reply:
Also fixed this differently in new code. Thank you for pointing this out.